### DISCUSSION TITLE
Using external DbContext within hook.

### DISCUSSION THREAD

**Discussion by gmullernh** [PiranhaCMS/piranha.core#1801] [Q&A] [answered]
Hey!
I'm adding a logging mechanism to a Piranha CMS instance. 
The ideia is to add an EF Core operation to another database context (that's being properly created) and hooking it on the `RegisterOnAfterSave` hook.
I'm getting a "**Cannot access a disposed object**" at the `logContext.Logs.Add` method.
After some [research](https://stackoverflow.com/questions/55543595/cannot-access-a-disposed-object-a-common-cause-of-this-error-is-disposing-a-con), found out that it might be being caused by a void method in the stack, that disposes the context before the operation. After some [more research](https://github.com/PiranhaCMS/piranha.core/blob/d1557674c1046558cdeee38d4e82f327aade9d90/core/Piranha/Runtime/HookManager.cs), I've found out that the delegates of Piranha's CMS Hook Manager, uses void methods.

Might this the issue?
Any ideias on how to trigger this hook using a Task or a method, without losing the context?

My implementation on the `Startup.cs`:
```C#
public void Configure(
    IApplicationBuilder app,
    IWebHostEnvironment env,
    IApi api,
    LogContext logContext,
    UserResolverService userResolverService) {
    // ...
    App.Init(api);
    // ...
    App.Hooks.Pages.RegisterOnAfterSave((page) => {
        logContext.Logs.Add(new LogChangeModel() {
          Id = page.Id.ToString(),
            Title = page.Title,
            Snapshot = JsonSerializer.Serialize(page),
            User = userResolverService.GetUser(),
            ChangeTime = DateTime.Now
        });
        logContext.Logs.SaveChanges();
      }
      // ...
    }
```

**Comment by tidyui** [comment, marked as answer]
The problem is that the context you’re using in the hook will be disposed as soon as the Configure method is done, you need to resolve the context inside the hook itself.

See this SO answer for reference! https://stackoverflow.com/a/65631856/1793832

Best regards

**Reply by gmullernh** [reply]
Thank you!
It worked like a charm :)
